src/devices/laserFromExternalPort/laserFromExternalPort.cpp

```diff
@@ -0,0 +1,216 @@
+{
+    mutex.lock();
+        data = lastScan;
+        stmp = lastStamp;
+    mutex.unlock();
+}
+
+//-------------------------------------------------------------------------------------
+
+bool LaserFromExternalPort::open(yarp::os::Searchable& config)
+{
+    Property subConfig;
+    m_info = "LaserFromExternalPort device";
+    m_port_name = "/laserFromExternalPort:i";
+
+#ifdef LASER_DEBUG
+    yDebug("%s\n", config.toString().c_str());
+#endif
+
+    if (this->parse(config) == false)
+    {
+        yError() << "LaserFromExternalPort: error parsing parameters";
+        return false;
+    }
+
+    yarp::os::Searchable& general_config = config.findGroup("SENSOR");
+    if (general_config.check("input_port_name"))
+       {m_port_name = general_config.find("input_port_name").asString();} //this parameter is optional
+
+    m_input_port.useCallback();
+    m_input_port.open(m_port_name);
+
+    PeriodicThread::start();
+
+    yInfo("LaserFromExternalPort: Sensor ready");
+    return true;
+}
+
+bool LaserFromExternalPort::close()
+{
+    PeriodicThread::stop();
+
+    m_input_port.close();
+
+    yInfo() << "LaserFromExternalPort closed";
+    return true;
+}
+
+
+
+bool LaserFromExternalPort::setDistanceRange(double min, double max)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    m_min_distance = min;
+    m_max_distance = max;
+    return true;
+}
+
+bool LaserFromExternalPort::setScanLimits(double min, double max)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    yWarning("setScanLimits not yet implemented");
+    return true;
+}
+
+
+
+bool LaserFromExternalPort::setHorizontalResolution(double step)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    yWarning("setHorizontalResolution not yet implemented");
+    return true;
+}
+
+bool LaserFromExternalPort::setScanRate(double rate)
+{
+    std::lock_guard<std::mutex> guard(m_mutex);
+    yWarning("setScanRate not yet implemented");
+    return false;
+}
+
+
+
+bool LaserFromExternalPort::threadInit()
+{
+#ifdef LASER_DEBUG
+    yDebug("LaserFromExternalPort:: thread initialising...\n");
+    yDebug("... done!\n");
+#endif
+
+    return true;
+}
+
+void LaserFromExternalPort::run()
+{
+#ifdef DEBUG_TIMING
+    double t1 = yarp::os::Time::now();
+#endif
+    std::lock_guard<std::mutex> guard(m_mutex);
+
+    m_input_port.getLast(m_last_scan_data, m_last_stamp);
+    size_t received_scans = m_last_scan_data.scans.size();
+
+    if (1)
+    {
+       //this ovverrides user setting with parameteres received from the port
+       m_sensorsNum = received_scans;
+       m_max_angle = m_last_scan_data.angle_max;
+       m_min_angle = m_last_scan_data.angle_min;
+       m_max_distance = m_last_scan_data.range_max;
+       m_min_distance = m_last_scan_data.range_min;
+       m_resolution   = received_scans/(m_max_angle - m_min_angle);
+       if (m_laser_data.size() != m_sensorsNum) m_laser_data.resize(m_sensorsNum);
+    }
+
+    double angle, distance;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'distance' can be reduced.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5040974)